src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -114,15 +116,22 @@ public string GetRuleSetFilePath(RuleSetGroup group)
         #endregion
 
         #region Public API
-        public void Initialize(IEnumerable<Project> projects)
+        public void Initialize(IEnumerable<Project> projects, Dictionary<LanguageGroup, QualityProfile> profilesMap)
         {
             if (projects == null)
             {
                 throw new ArgumentNullException(nameof(projects));
             }
 
+            if (profilesMap == null)
+            {
+                throw new ArgumentNullException(nameof(profilesMap));
+            }
+
             this.SolutionFullPath = this.projectSystem.GetCurrentActiveSolution().FullName;
 
+            profilesMap.ToList().ForEach(kv => qualityProfileMap.Add(kv.Key, kv.Value));
```

Comment:
What's the reason for creating and populating a new List<>? Just so you can use _.ForEach()_ rather than _foreach{}_?